### PR TITLE
Improve the clarity of the torch.Tensor.backward doc

### PR DESCRIPTION
**Pull request by lancerts**
Improve the clarity of the torch.Tensor.backward doc, particularly wrt the arg `gradient`. 
Reference https://pytorch.org/tutorials/beginner/blitz/autograd_tutorial.html, 
```
We need to explicitly pass a gradient argument in Q.backward() because it is a vector. gradient is a tensor of the same shape as Q, and it represents the gradient of Q w.r.t. itself
```

@janeyx99 feel free to assign to the corresponding reviewers, thanks

### CODE REVIEW

File: torch/_tensor.py [hunk 1]
```diff
@@ -491,12 +491,12 @@ def backward(
             See https://github.com/pytorch/pytorch/pull/60521#issuecomment-867061780 for more details.
 
         Args:
-            gradient (Tensor or None): Gradient w.r.t. the
-                tensor. If it is a tensor, it will be automatically converted
+            gradient (Tensor or None, optional): The gradient of the function
```
Comment (soulitzer) [inline torch/_tensor.py:494]:
```suggestion
            gradient (Tensor, optional): The gradient of the function
```

File: torch/_tensor.py [hunk 2]
```diff
@@ -491,12 +491,12 @@ def backward(
...
+                being differentiated w.r.t. ``self``.
+                If it is a tensor, it will be automatically converted
                 to a Tensor that does not require grad unless ``create_graph`` is True.
-                None values can be specified for scalar Tensors or ones that
-                don't require grad. If a None value would be acceptable then
-                this argument is optional.
+                None values can be specified for scalar tensors or tensors that
```
Comment (soulitzer) [inline torch/_tensor.py:498]: Since self must require grad, we can further simplify this to:
"This argument can be omitted if self is a scalar".

**Review by soulitzer — COMMENTED** [review]
Thanks for the updates, I added some small comments.

**Comment by lancerts** [comment]
> Thanks for the updates, I added some small comments.

@soulitzer updated, ty

**Review by soulitzer — APPROVED** [review]
Thanks!

**Pull request closed without merging**